packages/opentelemetry-core/src/internal/exporter.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { context } from '@opentelemetry/api';
+import { ExportResult } from '../ExportResult';
+import { suppressTracing } from '../trace/suppress-tracing';
+
+export interface Exporter<T> {
+  export(arg: T, resultCallback: (result: ExportResult) => void): void;
+}
+
+/**
+* Internal shared functionality used by Exporters while exporting data, including suppresion of Traces.
+*/
```

Comment:
Should we consider the https://typedoc.org/tags/experimental/ tag?

Reply:
That suggests the API may graduate from experimental in a near future. But internal utils may never get out of the internal scope.

Reply:
Then what about the `@ignore` tag so that it will not appear in documentation?

Reply:
The reason I suggest a dedicated namespace `internal` is that people tend to use everything that shows up in the IDE type-suggestions. Without the dedicated namespace, there is no significant difference between internal utils with the public interfaces in the IDE suggestions.

Reply:
I am not suggesting to remove the `internal` namespace, just trying to find ways to prevent it from showing up in documentation since it may not be clear from the generated docs that it is internal.

Reply:
Ah, I see. Since we didn't enable the tsc compiler option `stripInternal` (which is not recommended to be used anyways), I think https://typedoc.org/tags/internal/ is a good alternative.

Reply:
@hectorhdzg can you please apply the tag? rest of the PR looks good to me

Reply:
Tag should be there now, thanks for reviewing